File: July 2024/14-07-2024/Segregate 0s and 1s.py
#User function Template for python3

class Solution:
    def segregate0and1(self, arr):
        #Your Code Start
        l = 0
        r = len(arr) - 1
        while l<r:
            while arr[l] == 0 and l < r:
                l += 1
            while arr[r] == 1 and l<r:
                r -= 1
            if l < r:
                arr[l], arr[r] = arr[r], arr[l]
                l += 1
                r -= 1
        #Your Code End

#{ 
 # Driver Code Starts
class IntArray:

    def __init__(self) -> None:
        pass

    def Input(self, n):
        arr = [int(i) for i in input().strip().split()]  #array input
        return arr

    def Print(self, arr):
        for i in arr:
            print(i, end=" ")
        print()


if __name__ == "__main__":
    t = int(input())
    for _ in range(t):

        arr = list(map(int, input().strip().split()))

        obj = Solution()
        obj.segregate0and1(arr)

        print(*arr)

# } Driver Code Ends

File: July 2024/14-07-2024/Segregate 0s and 1s.js
//{ Driver Code Starts
// Initial Template for javascript

'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', inputStdin => { inputString += inputStdin; });

process.stdin.on('end', _ => {
    inputString =
        inputString.trim().split('\n').map(string => { return string.trim(); });

    main();
});

function readLine() { return inputString[currentLine++]; }

function main() {
    let t = parseInt(readLine());
    let i = 0;
    for (; i < t; i++) {
        // let n = parseInt(readLine());
        let arr = readLine().trim().split(" ").map((x) => parseInt(x));
        let n = arr.length;
        let obj = new Solution();
        let res = obj.segregate0and1(arr);
        let s = "";
        for (let j = 0; j < n; j++) {
            s += arr[j] + " ";
        }
        console.log(s);
    }
}
// } Driver Code Ends


// User function Template for javascript

/**
 * @param {number[]} arr

*/

class Solution {

    segregate0and1(arr) {

   //Your Code Start
        let l = 0;
        let r = arr.length -1;
        while (l<r){

            while(arr[l] === 0 && l<r){
                l += 1; 
            }
            
            while(arr[r]===1 && l<r){
                r -= 1;
            }
       
            if (l < r){
              let temp = arr[l];
                arr[l] = arr[r];
                arr[r] = temp;
                l += 1;
                r -= 1;
            }
      
    }
      //Your Code End
        
    }

}
